test/unit/rules/no-class-bindings-test.js

```diff
@@ -90,5 +94,340 @@ generateRuleTests({
         `);
       },
     },
+    {
+      template: '<SomeThing @classNameBindings={{@something}} />',
+
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          Array [
+            Object {
+              "column": 11,
+              "filePath": "layout.hbs",
+              "isFixable": false,
+              "line": 1,
+              "message": "Passing the \`@classNameBindings\` property as an argument within templates is not allowed.",
+              "moduleId": "layout",
+              "rule": "no-class-bindings",
+              "severity": 2,
+              "source": "@classNameBindings={{@something}}",
+            },
+          ]
+        `);
+      },
+    },
+
+    {
+      template: '<SomeThing @classNameBindings="isTruthy" />',
+      fixedTemplate: '<SomeThing class="{{if this.isTruthy "is-truthy"}}" />',
```

Comment:
I have two questions about this:

- wouldn't `isTruthy` refer to the `isTruthy` of `SomeThing` instead of the calling context (`this`)?
- I roughly remember that classNameBindings were allowed to be strings too, results in something like `{{if this.isTruthy (kebab-case this.isTruthy)}}`. Am I misremembering that?

Reply:
> wouldn't isTruthy refer to the isTruthy of SomeThing instead of the calling context (this)?

You are correct. I have introduced a bug in the fixer for this. 🤦 

> I roughly remember that classNameBindings were allowed to be strings too, results in something like {{if this.isTruthy (kebab-case this.isTruthy)}}. Am I misremembering that?

Yes, there are tests for this behavior already.